src/main/java/wl/contract/Retrieval.java

```diff
@@ -0,0 +1,30 @@
+package wl.contract;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Interface to retrieve objects.
+ *
+ * @param <T> Type of object to retrieve.
+ * @param <K> Type of object id.
+ */
+public interface Retrieval<T, K> {
```

Comment:
Could be `ID`, as we have in Sping Data JPA
```suggestion
public interface Retrieval<T, ID> {
```